src/cript/api/api.py

```diff
@@ -516,20 +520,22 @@ def search(
         # always putting a page parameter of 0 for all search URLs
         page_number = 0
 
+        api_endpoint: str = ""
         # requesting a page of some primary node
         if search_mode == SearchModes.NODE_TYPE:
-            api_endpoint: str = f"{self._host}/{node_type}"
+            api_endpoint = f"{self._host}/{node_type}"
 
         elif search_mode == SearchModes.CONTAINS_NAME:
-            api_endpoint: str = f"{self._host}/search/{node_type}"
+            api_endpoint = f"{self._host}/search/{node_type}"
 
         elif search_mode == SearchModes.EXACT_NAME:
-            api_endpoint: str = f"{self._host}/search/exact/{node_type}"
+            api_endpoint = f"{self._host}/search/exact/{node_type}"
 
         elif search_mode == SearchModes.UUID:
-            api_endpoint: str = f"{self._host}/{node_type}/{value_to_search}"
+            api_endpoint = f"{self._host}/{node_type}/{value_to_search}"
             # putting the value_to_search in the URL instead of a query
             value_to_search = None
 
+        assert api_endpoint != ""
```

Comment:
I like the way that this is handled, I doubt it'll ever be empty, but we should probably have something for the user in case they bump into an `assertion error` and have no idea where it came from and how to fix it

Reply:
we might want to write somethingin the docs, that if they ever encounter an assertion error, it is a bug and they should report it.

Reply:
I think putting it within docs can be good, but I bet we can put it as a good error, or just never hit this case. Something we can do for later